pandas/tests/indexes/datetimes/test_formats.py

```diff
@@ -45,3 +48,172 @@ def test_to_native_types():
 
     result = index.to_native_types(na_rep='pandas')
     tm.assert_numpy_array_equal(result, expected)
+
+
+class TestDatetimeIndexRendering(object):
+    def test_dti_repr_short(self):
+        dr = pd.date_range(start='1/1/2012', periods=1)
+        repr(dr)
+
+        dr = pd.date_range(start='1/1/2012', periods=2)
+        repr(dr)
+
+        dr = pd.date_range(start='1/1/2012', periods=3)
+        repr(dr)
+
+    def test_dti_representation(self):
+        idx = []
+        idx.append(DatetimeIndex([], freq='D'))
+        idx.append(DatetimeIndex(['2011-01-01'], freq='D'))
+        idx.append(DatetimeIndex(['2011-01-01', '2011-01-02'], freq='D'))
+        idx.append(DatetimeIndex(
+            ['2011-01-01', '2011-01-02', '2011-01-03'], freq='D'))
+        idx.append(DatetimeIndex(
+            ['2011-01-01 09:00', '2011-01-01 10:00', '2011-01-01 11:00'
+             ], freq='H', tz='Asia/Tokyo'))
+        idx.append(DatetimeIndex(
+            ['2011-01-01 09:00', '2011-01-01 10:00', pd.NaT], tz='US/Eastern'))
+        idx.append(DatetimeIndex(
+            ['2011-01-01 09:00', '2011-01-01 10:00', pd.NaT], tz='UTC'))
+
+        exp = []
+        exp.append("""DatetimeIndex([], dtype='datetime64[ns]', freq='D')""")
+        exp.append("DatetimeIndex(['2011-01-01'], dtype='datetime64[ns]', "
+                   "freq='D')")
+        exp.append("DatetimeIndex(['2011-01-01', '2011-01-02'], "
+                   "dtype='datetime64[ns]', freq='D')")
+        exp.append("DatetimeIndex(['2011-01-01', '2011-01-02', '2011-01-03'], "
+                   "dtype='datetime64[ns]', freq='D')")
+        exp.append("DatetimeIndex(['2011-01-01 09:00:00+09:00', "
+                   "'2011-01-01 10:00:00+09:00', '2011-01-01 11:00:00+09:00']"
+                   ", dtype='datetime64[ns, Asia/Tokyo]', freq='H')")
+        exp.append("DatetimeIndex(['2011-01-01 09:00:00-05:00', "
+                   "'2011-01-01 10:00:00-05:00', 'NaT'], "
+                   "dtype='datetime64[ns, US/Eastern]', freq=None)")
+        exp.append("DatetimeIndex(['2011-01-01 09:00:00+00:00', "
+                   "'2011-01-01 10:00:00+00:00', 'NaT'], "
+                   "dtype='datetime64[ns, UTC]', freq=None)""")
+
+        with pd.option_context('display.width', 300):
```

Comment:
looks fine. can you parameterize these (and others that look similar)

Reply:
Managed to parametrize a couple of for loops.  There is probably more than can be done in the next pass.